lldb/source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp

```diff
@@ -197,7 +197,64 @@ do {
         weak_self ? "Swift.Optional where Wrapped == " : "";
 
     // The expression text is inserted into the body of $__lldb_wrapped_expr_%u.
-    wrapped_stream.Printf(R"(
+    if (options.GetEvaluateSelfAsGeneric()) {
+      // When evaluating as generic, we have to bypass the Swift typesystem.
+      // We do this by having $_lldb_expr call a sink function which takes
+      // $__lldb_arg and 2 type erased parameters (the pointer to self, and the
+      // pointer to the metadata). After generating LLVM IR we redirect the call
+      // to $__lldb_generic_expr, which calls the method with the user's code.
+      // We need this extra function as we want a uniform signature for both
+      // reference and value types. We also need to make sure
+      // $__lldb_generic_expr and $__lldb_wrapped_expr_ are generic, and we do
+      // that by mentioning the generic parameters in the tuple argument of
+      // $__lldb_arg.
+      // FIXME: the current approach always passes in the first metadata
+      // pointer, change it so we can pass as many metadata pointers as
+      // necessary.
+      // FIXME: the current approach only evaluates self as generic, make it so
+      // we pass every variable in scope as generic, adding the variables and
+      // generic parameters to the signature as demanded.
+      // FIXME: the current approach names the generic parameter "T", use the
+      // user's name for the generic parameter, so they can refere to it in
+      // their expression.
+      wrapped_stream.Printf(
+          R"(
+extension %s$__lldb_context {
+  @LLDBDebuggerFunction %s
+  %s func $__lldb_wrapped_expr_%u<T>(_ $__lldb_arg: UnsafeMutablePointer<(T)>) {
+    %s
+  }
+}
+
+@LLDBDebuggerFunction %s
+func $__lldb_generic_expr<T>(_ $__lldb_arg: UnsafeMutablePointer<(T)>,
```

Comment:
It's unclear why only this is called "generic"; the above function is also generic. The difference is that this function is freestanding and extension method.

Reply:
What about `$__lldb_trampoline` to indicate it's purpose better?

Reply:
works for me